sklearn/cluster/mean_shift_.py

```diff
@@ -224,13 +235,79 @@ def get_bin_seeds(X, bin_size, min_bin_freq=1):
     return bin_seeds
 
 
+def _kernel_update(old_cluster_center, points, bandwidth, kernel, gamma):
+    """ Update cluster center according to kernel used
+
+    Parameters
+    ----------
+    old_window_center : array_like, shape=[n_features]
+        The location of the old centroid candidate.
+
+    points : array_like, shape=[n_samples, n_features]
+        All data points that fall within the window of which the new mean
+        is to be computed.
+
+    bandwidth : float
+        Size of the kernel. All points within this range are used to
+        compute the new mean. There should be no points outside this range
+        in the points variable.
+
+    kernel : string
+        The kernel used for updating the window center. Available are:
+        'flat', 'rbf', 'epanechnikov' and 'biweight'
+
+    gamma : float
+        Controls the width of the rbf kernel. Only used with rbf kernel
+        Lower values make it more like the flat kernel, higher values give
+        points closer to the old center more weights.
+
+    Notes
+    -----
+    Kernels as mentioned in the paper "Mean Shift, Mode Seeking, and
+    Clustering" by Yizong Cheng, published in IEEE Transaction On Pattern
+    Analysis and Machine Intelligence, vol. 17, no. 8, august 1995.
+
+    The rbf or Gaussian kernel is a truncated version, since in this
+    implementation of Mean Shift clustering only the points within the
```

Comment:
So the RBF kernel has bandwidth and gamma parameters? Is that how it is in the original paper?